### DISCUSSION TITLE
Use cast in migrations

### DISCUSSION THREAD

**Discussion by vdauchy** [spatie/laravel-settings#49] [Ideas]
Hi there,

Maybe I missed something but I seem that casts are not applied when running migrations. For example doing:
```
$this->migrator->inGroup(SomeSettings::group(), function (SettingsBlueprint $blueprint): void {
            $blueprint->add('some_duration', CarbonInterval::minutes(10));
        });
```
And having a proper 
```
'global_casts' => [
    CarbonInterval::class => CarbonIntervalCast::class,
    ...
],
```
Will not save the expected value returned by 'CarbonIntervalCast::set()'.

So I was wondering if there was any reason to this behavior and if no, would you be interested in a PR to add casts to migrations ?

**Comment by rubenvanassche** [comment]
Hi @vdauchy,

At the moment we do not have such functionality sadly enough, so if you could make a PR that would check for global casts within a migrations then I would certainly merge it 😄

**Reply by vdauchy** [reply]
Nice, will do as soon has I have the time :-)
